proposals/csharp-9.0/function-pointers.md

```diff
@@ -358,9 +358,12 @@ the attribute:
 
 * It is an error to directly call a method annotated with this attribute from C#. Users must obtain a function pointer to
 the method and then invoke that pointer.
-* It is an error to apply the attribute to anything other than a static method. The C# compiler will mark any non-static
-methods imported from metadata with this attribute as unsupported by the language.
-* It is an error to have managed types as parameters or the return type of a method marked with the attribute.
+* It is an error to apply the attribute to anything other than an ordinary static method or ordinary static local function.
+The C# compiler will mark any non-static methods imported from metadata with this attribute as unsupported by the language.
+* It is an error to have managed types as parameters or the return type of a method marked with the attribute. This includes
+types with type parameters, even if those type parameters are constrained to `unmanaged` or substituted with `unmanaged` types.
```

Comment:
Just want to check: Is `Vector128<int>` is allowed? (I assume that the answer is yes.)

Reply:
No, not by these rules. My understanding was that even something like this should be an error:
```cs
struct S2<T> where T : unmanaged { T Field; }
[UnmanagedCallersOnly]
void M(S2<int> param);
```

Reply:
There is nothing fundamental that prevents this from working and it works in .NET 5 today. I do not think we should be disallowing it in the compiler.

We have done work to enable blittable generic types in interop signatures in .NET 5: https://github.com/dotnet/runtime/pull/103